process/wsindexer/indexer.go

```diff
@@ -0,0 +1,100 @@
+package wsindexer
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go-core/websocketOutportDriver/data"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+var log = logger.GetOrCreate("process/wsindexer")
+
+type indexer struct {
+	marshaller marshal.Marshalizer
+	di         DataIndexer
+}
+
+func NewIndexer(marshaller marshal.Marshalizer, dataIndexer DataIndexer) (*indexer, error) {
```

Comment:
missing comments on exported items

Reply:
also, no nil checks. I would recommend to add them since the constructor is exported and it's not necessary called "safely" from within the package

Reply:
added